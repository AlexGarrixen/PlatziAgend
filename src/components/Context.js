import React, { createContext, useMemo } from 'react';
import {
  generateDaysOfWeek,
  generateTimeline,
  getMonthCurrent,
  getTimestamp,
} from '../utils';

export const Context = createContext({});

export const ContextProvider = ({
  children,
  startHour,
  endHour,
  unitDur,
  events,
}) => {
  const daysOfWeek = generateDaysOfWeek();
  const timeline = generateTimeline(startHour, endHour, unitDur);
  const monthName = getMonthCurrent();
  const timelineStart = getTimestamp(timeline[0].format24.split(' ')[0]);
  const timelineUnitDuration =
    getTimestamp(timeline[1].format24.split(' ')[0]) -
    getTimestamp(timeline[0].format24.split(' ')[0]);

  const contexValue = useMemo(
    () => ({
      daysOfWeek,
      timeline,
      monthName,
      events,
      timelineStart,
      timelineUnitDuration,
    }),
    [events, startHour, endHour, unitDur]
  );

  return <Context.Provider value={contexValue}>{children}</Context.Provider>;
};
