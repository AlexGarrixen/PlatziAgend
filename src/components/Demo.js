import React from 'react';
import Schedule from './index';

// example of events
const events = [
  {
    id: 1,
    title: 'Team Itsuki',
    date: '2020-09-07',
    hourStart: '15:30',
    hourEnd: '16:30',
  },
  {
    id: 2,
    title: 'Team Miku',
    date: '2020-09-09',
    hourStart: '13:30',
    hourEnd: '14:00',
  },
  {
    id: 3,
    title: 'Team Nino',
    date: '2020-09-10',
    hourStart: '14:30',
    hourEnd: '15:30',
  },
  {
    id: 4,
    title: 'Team ichika',
    date: '2020-09-12',
    hourStart: '15:30',
    hourEnd: '17:30',
  },
  {
    id: 5,
    title: 'Team youtsuba',
    date: '2020-09-07',
    hourStart: '14:00',
    hourEnd: '14:30',
  },
];

const Demo = () => (
  <section style={{ padding: '20px 0' }}>
    <div className='alert-message'>
      <img
        src='https://vignette.wikia.nocookie.net/5toubun-no-hanayome/images/d/d6/Miku_Nakano_Anime.png/revision/latest?cb=20190420141556'
        width={150}
        alt='picture'
      />
      <p>
        Hello developer this is a demo, feel free to be able to modify it and
        even improve it.
      </p>
    </div>
    <Schedule events={events} startHour={13} endHour={18} diffMinutUnit={30} />
    <div className='wrapper schedule__previewEvents'>
      <h4>Example Events</h4>
      <pre>
        <code>{JSON.stringify(events, null, 2)}</code>
      </pre>
    </div>
  </section>
);

export default Demo;
