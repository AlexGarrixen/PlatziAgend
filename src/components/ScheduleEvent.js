import React from 'react';
import PropTypes from 'prop-types';
import usePlaceEvent from '../hooks/usePlaceEvent';

const ScheduleEvent = ({ hourStart, hourEnd, title }) => {
  const { height, placement } = usePlaceEvent(hourStart, hourEnd);

  return (
    <div
      className='schedule__event'
      style={{
        height,
        top: placement,
      }}
    >
      {title}
    </div>
  );
};

ScheduleEvent.propTypes = {
  hourStart: PropTypes.string.isRequired,
  hourEnd: PropTypes.string.isRequired,
  title: PropTypes.string,
};

export default ScheduleEvent;
