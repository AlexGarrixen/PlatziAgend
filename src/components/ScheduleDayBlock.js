import React, { useContext, useMemo } from 'react';
import PropTypes from 'prop-types';
import { Context } from './Context';
import { isDateToday } from '../utils';
import EventItem from './ScheduleEvent';

const ScheduleDayBlock = ({ formatShort, formatDate }) => {
  const { events } = useContext(Context);
  const eventsOfDay = useMemo(
    () => events.filter((event) => isDateToday(event.date, formatDate)),
    [events]
  );

  return (
    <li className='schedule__dayBlock'>
      <div>{formatShort.toUpperCase()}</div>
      <div>
        {eventsOfDay.map((date) => (
          <EventItem key={date.id} {...date} />
        ))}
      </div>
    </li>
  );
};

ScheduleDayBlock.propTypes = {
  formatDate: PropTypes.string.isRequired,
  formatShort: PropTypes.string.isRequired,
};

export default ScheduleDayBlock;
