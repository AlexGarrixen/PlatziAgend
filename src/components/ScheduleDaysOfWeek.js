import React, { useContext } from 'react';
import { Context } from './Context';
import DayBlock from './ScheduleDayBlock';

const ScheduleDaysOfWeek = () => {
  const { daysOfWeek } = useContext(Context);

  return (
    <ul className='schedule__daysWeek'>
      {daysOfWeek.map(({ formatShort, formatDate }) => (
        <DayBlock
          formatShort={formatShort}
          formatDate={formatDate}
          key={formatDate}
        />
      ))}
    </ul>
  );
};

export default ScheduleDaysOfWeek;
