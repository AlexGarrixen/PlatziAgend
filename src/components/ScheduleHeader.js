import React, { useContext } from 'react';
import { Context } from './Context';

const ScheduleHeader = () => {
  const { monthName } = useContext(Context);

  return <div className='schedule__header'>{monthName.toUpperCase()}</div>;
};

export default ScheduleHeader;
