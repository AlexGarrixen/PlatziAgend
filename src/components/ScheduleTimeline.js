import React, { useContext } from 'react';
import { Context } from './Context';

const ScheduleTimeline = () => {
  const { timeline } = useContext(Context);

  return (
    <ul className='schedule__timeline'>
      {timeline.map(({ format12 }) => (
        <li key={format12} className='schedule__timeline-item'>
          {format12}
        </li>
      ))}
    </ul>
  );
};

export default ScheduleTimeline;
