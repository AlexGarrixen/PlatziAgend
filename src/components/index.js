import React from 'react';
import PropTypes from 'prop-types';
import { ContextProvider } from './Context';
import Header from './ScheduleHeader';
import Timeline from './ScheduleTimeline';
import DaysOfWeek from './ScheduleDaysOfWeek';

const Schedule = ({ startHour, endHour, events = [], diffMinutUnit }) => {
  return (
    <ContextProvider
      startHour={startHour}
      endHour={endHour}
      unitDur={diffMinutUnit}
      events={events}
    >
      <div className='wrapper'>
        <Header />
        <div className='schedule'>
          <Timeline />
          <DaysOfWeek />
        </div>
      </div>
    </ContextProvider>
  );
};

Schedule.propTypes = {
  startHour: PropTypes.number.isRequired,
  endHour: PropTypes.number.isRequired,
  diffMinutUnit: PropTypes.number,
  events: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      date: PropTypes.string,
      hourStart: PropTypes.string,
      hourEnd: PropTypes.string,
    })
  ),
};

export default Schedule;
