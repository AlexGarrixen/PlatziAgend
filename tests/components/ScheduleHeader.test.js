import React from 'react';
import { mount } from 'enzyme';
import { Context } from '../../src/components/Context';
import ScheduleHeader from '../../src/components/ScheduleHeader';

describe('<ScheduleHeader />', () => {
  const wrapper = mount(
    <Context.Provider value={{ monthName: 'September' }}>
      <ScheduleHeader />
    </Context.Provider>
  );

  test('Render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Value of month', () => {
    expect(wrapper.find('.schedule__header').text()).toBe('SEPTEMBER');
  });
});
