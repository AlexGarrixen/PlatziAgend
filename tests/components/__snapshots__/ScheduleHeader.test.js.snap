// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScheduleHeader /> Render 1`] = `
<div
  className="schedule__header"
>
  SEPTEMBER
</div>
`;
